include/aspect/particle/property/elastic_tensor_decomposition.h

```diff
@@ -0,0 +1,210 @@
+           * vectors has a finite dimension [...], i.e. using a different norm from eq. (2.3 will change distances
+           * but not the resulting decomposition.".
+           *
+           * With the three SCCS directions, the elastic tensor can be decomposed into the different
+           * symmetries in those three SCCS direction, that is, triclinic, monoclinic, orthorhombic,
+           * tetragonal, hexagonal, and isotropic (Browaeys & Chevrot, 2004).
+           *
+           * The dilatation_stiffness_tensor defines the stress to cause isotropic dilatation in the material.
+           * The voigt_stiffness_tensor defines the stress needed to cause an isotropic strain in the material
+           */
+          static
+          Tensor<2,3> compute_unpermutated_SCCS(const SymmetricTensor<2,3> &dilatation_stiffness_tensor,
+                                                const SymmetricTensor<2,3> &voigt_stiffness_tensor);
+
+          /**
+           * Uses the Symmetry Cartesian Coordinate System (SCCS) to try the different permutations to
+           * determine what is the best projections.
+           * This is based on Browaeys and Chevrot (2004), GJI (doi: 10.1111/j.1365-246X.2004.024115.x),
+           * which states at the end of paragraph 3.3 that "... an important property of an orthogonal projection
+           * is that the distance between a vector $X$ and its orthogonal projection $X_H = p(X)$ on a given
+           * subspace is minimum. These two features ensure that the decomposition is optimal once a 3-D Cartesian
+           * coordinate system is chosen.". The other property they talk about is that "The space of elastic
+           * vectors has a finite dimension [...], i.e. using a different norm from eq. (2.3 will change distances
+           * but not the resulting decomposition.".
+           */
+          static
+          std::array<std::array<double,3>,7>
+          compute_elastic_tensor_SCCS_decompositions(
+            const Tensor<2,3> &unpermutated_SCCS,
+            const SymmetricTensor<2,6> &elastic_matrix);
+
+          /**
+           * Set up the information about the names and number of components
+           * this property requires.
+           *
+           * @return A vector that contains pairs of the property names and the
+           * number of components this property plugin defines.
+           */
+          std::vector<std::pair<std::string, unsigned int>>
+          get_property_information() const  override;
+
+          /**
+           * Declare parameters
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Parse parameters
+           */
+          void
+          parse_parameters (ParameterHandler &prm) override;
+
+
+          /**
+           * The tenors below can be used to project matrices to different symmetries.
```

Comment:
```suggestion
           * The tensors below can be used to project matrices to different symmetries.
```

Also please spell out the different symmetry systems. Below you only use abbreviations.